### PR TITLE
doc: net: Add socket offloading documentation

### PR DESCRIPTION
**Pull request by rlubos**
Add brief documentation of socket offloading

### CODE REVIEW

**Review by jukkar — APPROVED** [review]
Thanks Robert! LGTM

File: doc/reference/networking/net_offload.rst [hunk 1]
```diff
@@ -16,3 +16,23 @@ API Reference
...
 .. doxygengroup:: net_offload
    :project: Zephyr
+
+Socket Offloading
+#################
+
+Overview
+********
+
+Beside of the network offloading API, Zephyr allows to offload networking
```
Comment (GAnthony) [inline doc/reference/networking/net_offload.rst]: "In addition to the network offloading API, Zephyr allows offloading ..."

File: doc/reference/networking/net_offload.rst [hunk 2]
```diff
@@ -16,3 +16,23 @@ API Reference
...
+functionality at the socket API level. With this approach, vendors that provide
+alternate implementation of the networking stack, exposing socket API for their
+networking devices, can easily integrate them with Zephyr.
```
Comment (GAnthony) [inline doc/reference/networking/net_offload.rst]: ", vendors who provide an alternate implementation of the networking stack, "

", can easily integrate it with Zephyr".

**Review by GAnthony — APPROVED** [review]
Many thanks for adding some documentation!
 Just a few minor suggestions...

**Comment by rlubos** [comment]
@GAnthony Thank you, I've pushed updated version.

**Review by dbkinder — APPROVED** [review]
+1 for docs

**Pull request merged**